src/chart/tree/TreeView.js

```diff
@@ -27,6 +27,85 @@ import View from '../../coord/View';
 import * as roamHelper from '../../component/helper/roamHelper';
 import RoamController from '../../component/helper/RoamController';
 import {onIrrelevantElement} from '../../component/helper/cursorHelper';
+import { __DEV__ } from '../../config';
+import {parsePercent} from '../../util/number';
+
+var TreeShape = graphic.extendShape({
+    shape: {
+        parentPoint: [],
+        childPoints: [],
+        orient: '',
+        forkPosition: ''
+    },
+
+    buildPath: function (ctx, shape) {
+        var ptMin = [Infinity, Infinity];
+        var ptMax = [-Infinity, -Infinity];
+        var points = shape.childPoints;
+        for (var i = 0; i < points.length; i++) {
```

Comment:
I think `ptMin` `ptMax` and the `switch case` of 'TB' 'BT' 'LR' 'RL' and `computeMidPoints ` are not needed. 
The code can be simplified if you wish. And some critical case need to be considered. For example:

```js
buildPath: function (ctx, shape) {
    var childPoints = shape.childPoints;
    var parentPoint = shape.parentPoint;
    var childLen = childPoints.length;
    var firstChildPos = childPoints[1];
    var lastChildPos = childPoints[childLen - 1];

    if (!childLen || !parentPoint) {
        return;
    }

    if (childLen === 1) {
        ctx.moveTo(firstChildPos[0], firstChildPos[1]);
        ctx.lineTo(parentPoint[0], parentPoint[1]);
        return;
    }

    var forkDim = (orient === 'TB' || orient === 'BT') ? 1 : 0;
    var otherDim = 1 - forkDim;
    var forkPosition = parsePercent(shape.forkPosition, 1);
    var tmpPt = [];

    ctx.moveTo(parentPoint[0], parentPoint[1]);
    tmpPt[forkDim] = parentPoint[forkDim] + (firstChildPos[forkDim] - parentPoint[forkDim]) * forkPosition;
    tmpPt[otherDim] = parentPoint[otherDim];
    ctx.lineTo(tmpPt[0], tmpPt[1]);

    ctx.moveTo(firstChildPos[0], firstChildPos[1]);
    tmpPt[otherDim] = firstChildPos[otherDim];
    ctx.lineTo(tmpPt[0], tmpPt[1]);
    tmpPt[otherDim] = lastChildPos[otherDim];
    ctx.lineTo(tmpPt[0], tmpPt[1]);
    ctx.lineTo(lastChildPos[0], lastChildPos[1]);

    for (var i = 1; i < childLen - 1; i++) {
        var childPos = childPoints[i];
        ctx.moveTo(childPos[0], childPos[1]);
        tmpPt[otherDim] = childPos[otherDim];
        ctx.lineTo(tmpPt[0], tmpPt[1]);
    }
}
```